flask_restful/__init__.py

```diff
@@ -417,20 +430,28 @@ def url_for(self, resource, **values):
     def make_response(self, data, *args, **kwargs):
         """Looks up the representation transformer for the requested media
         type, invoking the transformer to create a response object. This
-        defaults to (application/json) if no transformer is found for the
-        requested mediatype.
+        defaults to default_mediatype if no transformer is found for the
+        requested mediatype. If default_mediatype is None, a 406 Not 
+        Acceptable response will be sent as per RFC 2616 section 14.1
 
         :param data: Python object containing response data to be transformed
         """
-        for mediatype in self.mediatypes() + [self.default_mediatype]:
-            if mediatype in self.representations:
-                resp = self.representations[mediatype](data, *args, **kwargs)
-                resp.headers['Content-Type'] = mediatype
-                return resp
+        default_mediatype = kwargs.pop('fallback_mediatype', None) or self.default_mediatype
+        mediatype = request.accept_mimetypes.best_match(
+            self.representations, 
+            default=default_mediatype,
+        )
+        if mediatype is None:
+            raise NotAcceptable()
+        if mediatype in self.representations:
```

Comment:
With the current PR code, I think the only way to trigger the situation you describe would be by having `self.default_mediatype` set to something for which there is no representation in `self.representations` - the `best_match` function will otherwise ensure that only media types that have representations are included. The only reason for keeping the original guard code `if mediatype in self.representations:` is in case the user explicitly sets `default_mediatype` to be a media type for which it has no representation. 

I believe what would happen in that case is that instead of raising a `NotAcceptable` exception triggering the 406 http status code, `make_response` would return `None` instead of a response object as you suggest - the end result of which would be a generic `InternalServerError` (500). 

Given that the underlying issue in that case is one of an internal misconfiguration (the Flask-restful API defines a default mediatype for which it has no representation), I think the 500 status is probably more appropriate than the 406 status, but I don't feel strongly about that point, so if you'd like me to change it, I can do that. Another option might be to raise an `InternalServerError` explicitly and include the more specific problem in the description text. Along those lines, I also considered including the list of valid representations in the `NotAcceptable` description text - do you think that would be useful?